### ISSUE TITLE
Crash when joining multiplayer game

### ISSUE THREAD

**Issue by angrykirc** [opennox/opennox#693]
Something goes wrong when connecting to (presumably) vanilla server
version: v1.9.0-alpha13
```
2024/05/19 22:30:49 [game]: client connected: nox_xxx_gameChangeMap_43DEB0, gameloop.go:847
2024/05/19 22:30:49 game flag unset: Flag21 | Flag24
2024/05/19 22:30:49 [render]: recreating surface: 640x480 -> 1024x768
2024/05/19 22:30:49 [console]: system> You already know that ability.
2024/05/19 22:30:49 [console]: system> You already know that ability.
2024/05/19 22:30:49 [console]: system> You already know that ability.
2024/05/19 22:30:49 [console]: system> You already know that ability.
2024/05/19 22:30:49 [console]: system> You already know that ability.
2024/05/19 22:30:49 [console]: system> You already know that ability.
2024/05/19 22:30:49 [console]: system> You already know that ability.
2024/05/19 22:30:49 [console]: system> You already know that ability.
2024/05/19 22:30:49 [console]: system> You already know that ability.
2024/05/19 22:30:49 [console]: system> You already know that ability.
panic: overflow when handling net packet: [15:8]
PACKET: (1022 bytes)
27be9da4ff000000007c0c66080000ff80007a041899ffb20f8d040b45fff10e75043a0bfffc0f8c04e9e9ff040f8b04e9e9ff6f0fc902d245400000ff0200c902070d3b0b4000000000a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900391581003911800db20c80103974800db20b8000391a81003971800d390f800d3912800db20f8110b20e81003910800db29580185e020000a70045c8a70024cba7002dcba70030cba7004acbaacc000002ec01cc010003cd0105cc020003cd0205cc030003cd0405cc040003cd0505cc050003cd0305eb06cc060005eab69d1302cc07003cb1016d6e617661756c7400536570686972612053657276650000ffffffffffefffffffffffffffffffffffffffffffffffffffffffff07210f000000e40101cc0800812dca0a4100720063006f00780000000000000000000000000000000000000000000000000000000000000000000000000000000000085200007d00000000000000000000000000a28164a2816409000c09000ca281640c071507150000000000000000000100040005000001000200000000000000000146657269636974313100cc0900074fca8a00000001cc0a000750ca8a00020000cc0b000b52ca8a01000000c8baffffcc0c000b52ca8a04000000ffbfffffcc0d00812d720645006e00670065006c00000000000000000000000000000000000000000000000000000000000000000000000000000000005c4400002300000000000000000000000100341900da9a6eda9a6eda9a6eda9a6e141f14141500000000000000000005000100444400000000100000000000000001496b6f00636974313100cc0e000b51728600001000ffff9cffcc0f000b52728640000000ffff90ffcc1000074f728600400000cc11000b52728604000000ffc7ffffcc12000b52728600040000ffd2c7ffcc1300812dc01170006500740072006f0000000000000000000000000000000000000000000000000000000000000000000000000000000000444800003700000000000000000000000200381d68e4e4c1e4e4c1e4e4c1e4e4c11515151515000000000000000000fbff0300050410000a000000000000000000017065746572696f003100cc14000750c09102000000cc15000750c09108000000cc16000b52c09100
SPLIT:
27be9d
a4ff000000007c0c66080000ff80007a041899ffb20f8d040b45fff10e75043a0bfffc0f8c04e9e9ff040f8b04e9e9ff6f0fc902d245400000ff0200c902070d3b0b4000000000
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
39158100
3911800d
b20c8010
3974800d
b20b8000
391a8100
3971800d
390f800d
3912800d
b20f8110
b20e8100
3910800d
b2958018
5e020000
a70045c8
a70024cb
a7002dcb
a70030cb
a7004acb
aa
cc000002ec01
cc010003cd0105
cc020003cd0205
cc030003cd0405
cc040003cd0505
cc050003cd0305
eb06
cc060005eab69d1302
cc07003cb1016d6e617661756c7400536570686972612053657276650000ffffffffffefffffffffffffffffffffffffffffffffffffffffffff07210f000000
e40101
cc0800812dca0a4100720063006f00780000000000000000000000000000000000000000000000000000000000000000000000000000000000085200007d00000000000000000000000000a28164a2816409000c09000ca281640c071507150000000000000000000100040005000001000200000000000000000146657269636974313100
cc0900074fca8a00000001
cc0a000750ca8a00020000
cc0b000b52ca8a01000000c8baffff
cc0c000b52ca8a04000000ffbfffff
cc0d00812d720645006e00670065006c00000000000000000000000000000000000000000000000000000000000000000000000000000000005c4400002300000000000000000000000100341900da9a6eda9a6eda9a6eda9a6e141f14141500000000000000000005000100444400000000100000000000000001496b6f00636974313100
cc0e000b51728600001000ffff9cff
cc0f000b52728640000000ffff90ff
cc1000074f728600400000
cc11000b52728604000000ffc7ffff
cc12000b52728600040000ffd2c7ff
cc1300812dc01170006500740072006f0000000000000000000000000000000000000000000000000000000000000000000000000000000000444800003700000000000000000000000200381d68e4e4c1e4e4c1e4e4c1e4e4c11515151515000000000000000000fbff0300050410000a000000000000000000017065746572696f003100
cc14000750c09102000000
cc15000750c09108000000
cc16000b52c09100
 [recovered]
	panic: overflow when handling net packet: [15:8]
PACKET: (1022 bytes)
27be9da4ff000000007c0c66080000ff80007a041899ffb20f8d040b45fff10e75043a0bfffc0f8c04e9e9ff040f8b04e9e9ff6f0fc902d245400000ff0200c902070d3b0b4000000000a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900391581003911800db20c80103974800db20b8000391a81003971800d390f800d3912800db20f8110b20e81003910800db29580185e020000a70045c8a70024cba7002dcba70030cba7004acbaacc000002ec01cc010003cd0105cc020003cd0205cc030003cd0405cc040003cd0505cc050003cd0305eb06cc060005eab69d1302cc07003cb1016d6e617661756c7400536570686972612053657276650000ffffffffffefffffffffffffffffffffffffffffffffffffffffffff07210f000000e40101cc0800812dca0a4100720063006f00780000000000000000000000000000000000000000000000000000000000000000000000000000000000085200007d00000000000000000000000000a28164a2816409000c09000ca281640c071507150000000000000000000100040005000001000200000000000000000146657269636974313100cc0900074fca8a00000001cc0a000750ca8a00020000cc0b000b52ca8a01000000c8baffffcc0c000b52ca8a04000000ffbfffffcc0d00812d720645006e00670065006c00000000000000000000000000000000000000000000000000000000000000000000000000000000005c4400002300000000000000000000000100341900da9a6eda9a6eda9a6eda9a6e141f14141500000000000000000005000100444400000000100000000000000001496b6f00636974313100cc0e000b51728600001000ffff9cffcc0f000b52728640000000ffff90ffcc1000074f728600400000cc11000b52728604000000ffc7ffffcc12000b52728600040000ffd2c7ffcc1300812dc01170006500740072006f0000000000000000000000000000000000000000000000000000000000000000000000000000000000444800003700000000000000000000000200381d68e4e4c1e4e4c1e4e4c1e4e4c11515151515000000000000000000fbff0300050410000a000000000000000000017065746572696f003100cc14000750c09102000000cc15000750c09108000000cc16000b52c09100
SPLIT:
27be9d
a4ff000000007c0c66080000ff80007a041899ffb20f8d040b45fff10e75043a0bfffc0f8c04e9e9ff040f8b04e9e9ff6f0fc902d245400000ff0200c902070d3b0b4000000000
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
39158100
3911800d
b20c8010
3974800d
b20b8000
391a8100
3971800d
390f800d
3912800d
b20f8110
b20e8100
3910800d
b2958018
5e020000
a70045c8
a70024cb
a7002dcb
a70030cb
a7004acb
aa
cc000002ec01
cc010003cd0105
cc020003cd0205
cc030003cd0405
cc040003cd0505
cc050003cd0305
eb06
cc060005eab69d1302
cc07003cb1016d6e617661756c7400536570686972612053657276650000ffffffffffefffffffffffffffffffffffffffffffffffffffffffff07210f000000
e40101
cc0800812dca0a4100720063006f00780000000000000000000000000000000000000000000000000000000000000000000000000000000000085200007d00000000000000000000000000a28164a2816409000c09000ca281640c071507150000000000000000000100040005000001000200000000000000000146657269636974313100
cc0900074fca8a00000001
cc0a000750ca8a00020000
cc0b000b52ca8a01000000c8baffff
cc0c000b52ca8a04000000ffbfffff
cc0d00812d720645006e00670065006c00000000000000000000000000000000000000000000000000000000000000000000000000000000005c4400002300000000000000000000000100341900da9a6eda9a6eda9a6eda9a6e141f14141500000000000000000005000100444400000000100000000000000001496b6f00636974313100
cc0e000b51728600001000ffff9cff
cc0f000b52728640000000ffff90ff
cc1000074f728600400000
cc11000b52728604000000ffc7ffff
cc12000b52728600040000ffd2c7ff
cc1300812dc01170006500740072006f0000000000000000000000000000000000000000000000000000000000000000000000000000000000444800003700000000000000000000000200381d68e4e4c1e4e4c1e4e4c1e4e4c11515151515000000000000000000fbff0300050410000a000000000000000000017065746572696f003100
cc14000750c09102000000
cc15000750c09108000000
cc16000b52c09100


goroutine 1 [running, locked to thread]:
github.com/noxworld-dev/opennox/v1.RunArgs.func1()
	/github/workspace/src/main.go:96 +0x9b
panic({0x8e56120, 0xa42a930})
	/go/src/runtime/panic.go:920 +0x24c
github.com/noxworld-dev/opennox/v1.(*Client).nox_xxx_netOnPacketRecvCli48EA70(0xaf00000, 0x1f, {0xa54a902, 0x3fe, 0x8fe})
	/github/workspace/src/network.go:1062 +0x6d8
github.com/noxworld-dev/opennox/v1.nox_xxx_netHandleCliPacket_43C860({0xa52ca80, 0x0}, {0xa54a902, 0x3fe, 0x8fe})
	/github/workspace/src/network.go:1083 +0xae
github.com/noxworld-dev/opennox/v1/internal/netstr.(*conn).callOnReceive(...)
	/github/workspace/src/internal/netstr/netstr.go:559
github.com/noxworld-dev/opennox/v1/internal/netstr.Handle.RecvLoop({0xa52ca80, 0x0}, 0x1)
	/github/workspace/src/internal/netstr/netstr.go:1520 +0x7a2
github.com/noxworld-dev/opennox/v1.sub_43CCA0()
	/github/workspace/src/client_only.go:62 +0x6e
github.com/noxworld-dev/opennox/v1.(*Client).nox_xxx_client_435F80_draw(0xaf00000)
	/github/workspace/src/client_draw.go:125 +0x1a4
github.com/noxworld-dev/opennox/v1.(*Client).clientDraw(...)
	/github/workspace/src/video_client.go:61
github.com/noxworld-dev/opennox/v1.(*Client).Update(0xaf00000)
	/github/workspace/src/client.go:141 +0x6a
github.com/noxworld-dev/opennox/v1.mainloop_43E290(0x1)
	/github/workspace/src/gameloop.go:161 +0x543
github.com/noxworld-dev/opennox/v1.mainloop_43E290(0x0)
	/github/workspace/src/gameloop.go:186 +0x743
github.com/noxworld-dev/opennox/v1.cmainLoop()
	/github/workspace/src/gameloop.go:354 +0x13e
github.com/noxworld-dev/opennox/v1.RunArgs({0xa4d4058, 0x1, 0x1})
	/github/workspace/src/main.go:450 +0x2ae3
main.main()
	cmd/opennox/main.go:12 +0x3b
```

**Comment by angrykirc** [comment]
also crashes with binary built from the dev branch [b4e4299c], albeit with a slightly different stacktrace
```
panic: overflow when handling net packet: [4:1]
PACKET: (1022 bytes)
277045a4ff000000000009aa0d0000ff2d12740452e6ffe70e8902bff7ff671089021409ff3f12890205f0ff820f89021115ff7206c90286426004ff6f04c9023fd4400000ff0200c9026609430c4000ff040d7702c29100ffb9067e020d0a8a0e000000a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900b23d8018a70045c8a70024cba7002dcba70030cba7004acbaacc000002ec01cc010003cd0105cc020003cd0205cc030003cd0405cc040003cd0505cc050003cd0305eb06cc060005ea68451402cc07003cb101657374617465006200536570686972612053657276650000ffffffffffefffffffffffffffffffffffffffffffffffffffffffff07210f000000e40101cc0000079e0e00ca0a2204cc0800812dca0a4100720063006f00780000000000000000000000000000000000000000000000000000000000000000000000000000000000085200007d00000000000000000000000000a28164a2816409000c09000ca281640c071507150000000000000000000500090005024200000400000000000000000146657269636974313100cc09000750ca8a00040000cc0a00074fca8a00004000cc0b00074fca8a00020000cc0c00074fca8a00000200cc0d000b52ca8a01000000c8baffffcc0e000b52ca8a04000000ffbfffffcc0f00812d720645006e00670065006c00000000000000000000000000000000000000000000000000000000000000000000000000000000005c4400002300000000000000000000000100341900da9a6eda9a6eda9a6eda9a6e141f14141500000000000000000005000600054400000000000000000000000001496b6f00636974313100cc1000074f728600400000cc11000b52728601000000c8c7ffffcc12000b52728604000000ffc7ffffcc13000b52728600040000ffd2c7ffcc1400812dce074a00610063006b00000000000000000000000000000000000000000000000000000000000000000000000000000000000000444800003700000000000000000000000200734d22da9a6eda9a6eda9a6eda9a6e0c071317060020003200000000000e000400050400000a0000000000000000000164656d747269656f7000cc15000750ce8702000000cc16000750ce8708000000cc17000b52ce8701000000b9caffffcc
SPLIT:
277045
a4ff000000000009aa0d0000ff2d12740452e6ffe70e8902bff7ff671089021409ff3f12890205f0ff820f89021115ff7206c90286426004ff6f04c9023fd4400000ff0200c9026609430c4000ff040d7702c29100ffb9067e020d0a8a0e000000
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
b23d8018
a70045c8
a70024cb
a7002dcb
a70030cb
a7004acb
aa
cc000002ec01
cc010003cd0105
cc020003cd0205
cc030003cd0405
cc040003cd0505
cc050003cd0305
eb06
cc060005ea68451402
cc07003cb101657374617465006200536570686972612053657276650000ffffffffffefffffffffffffffffffffffffffffffffffffffffffff07210f000000
e40101
cc0000079e0e00ca0a2204
cc0800812dca0a4100720063006f00780000000000000000000000000000000000000000000000000000000000000000000000000000000000085200007d00000000000000000000000000a28164a2816409000c09000ca281640c071507150000000000000000000500090005024200000400000000000000000146657269636974313100
cc09000750ca8a00040000
cc0a00074fca8a00004000
cc0b00074fca8a00020000
cc0c00074fca8a00000200
cc0d000b52ca8a01000000c8baffff
cc0e000b52ca8a04000000ffbfffff
cc0f00812d720645006e00670065006c00000000000000000000000000000000000000000000000000000000000000000000000000000000005c4400002300000000000000000000000100341900da9a6eda9a6eda9a6eda9a6e141f14141500000000000000000005000600054400000000000000000000000001496b6f00636974313100
cc1000074f728600400000
cc11000b52728601000000c8c7ffff
cc12000b52728604000000ffc7ffff
cc13000b52728600040000ffd2c7ff
cc1400812dce074a00610063006b00000000000000000000000000000000000000000000000000000000000000000000000000000000000000444800003700000000000000000000000200734d22da9a6eda9a6eda9a6eda9a6e0c071317060020003200000000000e000400050400000a0000000000000000000164656d747269656f7000
cc15000750ce8702000000
cc16000750ce8708000000
cc17000b52ce8701000000b9caffff
cc
 [recovered]
	panic: overflow when handling net packet: [4:1]
PACKET: (1022 bytes)
277045a4ff000000000009aa0d0000ff2d12740452e6ffe70e8902bff7ff671089021409ff3f12890205f0ff820f89021115ff7206c90286426004ff6f04c9023fd4400000ff0200c9026609430c4000ff040d7702c29100ffb9067e020d0a8a0e000000a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900a90f007573652e633a4861644162696c69747900b23d8018a70045c8a70024cba7002dcba70030cba7004acbaacc000002ec01cc010003cd0105cc020003cd0205cc030003cd0405cc040003cd0505cc050003cd0305eb06cc060005ea68451402cc07003cb101657374617465006200536570686972612053657276650000ffffffffffefffffffffffffffffffffffffffffffffffffffffffff07210f000000e40101cc0000079e0e00ca0a2204cc0800812dca0a4100720063006f00780000000000000000000000000000000000000000000000000000000000000000000000000000000000085200007d00000000000000000000000000a28164a2816409000c09000ca281640c071507150000000000000000000500090005024200000400000000000000000146657269636974313100cc09000750ca8a00040000cc0a00074fca8a00004000cc0b00074fca8a00020000cc0c00074fca8a00000200cc0d000b52ca8a01000000c8baffffcc0e000b52ca8a04000000ffbfffffcc0f00812d720645006e00670065006c00000000000000000000000000000000000000000000000000000000000000000000000000000000005c4400002300000000000000000000000100341900da9a6eda9a6eda9a6eda9a6e141f14141500000000000000000005000600054400000000000000000000000001496b6f00636974313100cc1000074f728600400000cc11000b52728601000000c8c7ffffcc12000b52728604000000ffc7ffffcc13000b52728600040000ffd2c7ffcc1400812dce074a00610063006b00000000000000000000000000000000000000000000000000000000000000000000000000000000000000444800003700000000000000000000000200734d22da9a6eda9a6eda9a6eda9a6e0c071317060020003200000000000e000400050400000a0000000000000000000164656d747269656f7000cc15000750ce8702000000cc16000750ce8708000000cc17000b52ce8701000000b9caffffcc
SPLIT:
277045
a4ff000000000009aa0d0000ff2d12740452e6ffe70e8902bff7ff671089021409ff3f12890205f0ff820f89021115ff7206c90286426004ff6f04c9023fd4400000ff0200c9026609430c4000ff040d7702c29100ffb9067e020d0a8a0e000000
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
a90f007573652e633a4861644162696c69747900
b23d8018
a70045c8
a70024cb
a7002dcb
a70030cb
a7004acb
aa
cc000002ec01
cc010003cd0105
cc020003cd0205
cc030003cd0405
cc040003cd0505
cc050003cd0305
eb06
cc060005ea68451402
cc07003cb101657374617465006200536570686972612053657276650000ffffffffffefffffffffffffffffffffffffffffffffffffffffffff07210f000000
e40101
cc0000079e0e00ca0a2204
cc0800812dca0a4100720063006f00780000000000000000000000000000000000000000000000000000000000000000000000000000000000085200007d00000000000000000000000000a28164a2816409000c09000ca281640c071507150000000000000000000500090005024200000400000000000000000146657269636974313100
cc09000750ca8a00040000
cc0a00074fca8a00004000
cc0b00074fca8a00020000
cc0c00074fca8a00000200
cc0d000b52ca8a01000000c8baffff
cc0e000b52ca8a04000000ffbfffff
cc0f00812d720645006e00670065006c00000000000000000000000000000000000000000000000000000000000000000000000000000000005c4400002300000000000000000000000100341900da9a6eda9a6eda9a6eda9a6e141f14141500000000000000000005000600054400000000000000000000000001496b6f00636974313100
cc1000074f728600400000
cc11000b52728601000000c8c7ffff
cc12000b52728604000000ffc7ffff
cc13000b52728600040000ffd2c7ff
cc1400812dce074a00610063006b00000000000000000000000000000000000000000000000000000000000000000000000000000000000000444800003700000000000000000000000200734d22da9a6eda9a6eda9a6eda9a6e0c071317060020003200000000000e000400050400000a0000000000000000000164656d747269656f7000
cc15000750ce8702000000
cc16000750ce8708000000
cc17000b52ce8701000000b9caffff
cc


goroutine 1 [running, locked to thread]:
github.com/noxworld-dev/opennox/v1.RunArgs.func1()
	/home/runner/opennox/main.go:96 +0x9b
panic({0x8ee8480, 0xf1bd9f0})
	/usr/local/go/src/runtime/panic.go:920 +0x24c
github.com/noxworld-dev/opennox/v1.(*Client).nox_xxx_netOnPacketRecvCli48EA70(0xb692000, 0x1f, {0xad43202, 0x3fe, 0x8fe})
	/home/runner/opennox/network_client.go:484 +0x6d8
github.com/noxworld-dev/opennox/v1.nox_xxx_netHandleCliPacket_43C860({0x92e7d5c, 0xad0ce10}, {0xad43202, 0x3fe, 0x8fe})
	/home/runner/opennox/network_client.go:41 +0xae
github.com/noxworld-dev/opennox/v1/internal/netstr.(*Conn).callOnReceive(...)
	/home/runner/opennox/internal/netstr/conn.go:105
github.com/noxworld-dev/opennox/v1/internal/netstr.(*Conn).recvLoop(0xad0ce10, 0x1)
	/home/runner/opennox/internal/netstr/conn.go:669 +0x790
github.com/noxworld-dev/opennox/v1/internal/netstr.(*Conn).RecvLoop(...)
	/home/runner/opennox/internal/netstr/conn.go:544
github.com/noxworld-dev/opennox/v1.sub_43CCA0()
	/home/runner/opennox/client_only.go:61 +0x6e
github.com/noxworld-dev/opennox/v1.(*Client).nox_xxx_client_435F80_draw(0xb692000)
	/home/runner/opennox/client_draw.go:124 +0x1c9
github.com/noxworld-dev/opennox/v1.(*Client).clientDraw(...)
	/home/runner/opennox/video_client.go:61
github.com/noxworld-dev/opennox/v1.(*Client).Update(0xb692000)
	/home/runner/opennox/client.go:145 +0x6a
github.com/noxworld-dev/opennox/v1.mainloop_43E290(0x1)
	/home/runner/opennox/gameloop.go:161 +0x543
github.com/noxworld-dev/opennox/v1.mainloop_43E290(0x0)
	/home/runner/opennox/gameloop.go:186 +0x743
github.com/noxworld-dev/opennox/v1.cmainLoop()
	/home/runner/opennox/gameloop.go:354 +0x13e
github.com/noxworld-dev/opennox/v1.RunArgs({0xacd4058, 0x1, 0x1})
	/home/runner/opennox/main.go:461 +0x2c17
main.main()
	cmd/opennox/main.go:12 +0x3b

```

**Comment by angrykirc** [comment]
The bad packet here is cc1600**0b52**c09100, which is a "MSG_REPORT_MODIFIABLE_ARMOR_EQUIP" message embedded into a "MSG_SEQ_IMPORTANT" packet, and indeed it should be 7 bytes longer in first case, and much longer in second (there's complete packet body missing). Currently I'm trying to reconstruct the sequence of actions that lead to this
Worth noting that I've only been able to reproduce this bug if server is vanilla, and there is 3+ players on the server

**Comment by angrykirc** [comment]
Fixed by 28eab2aba4116f216d2933cca67dcb31fbfe7fa0
